src/linker/Linker.Steps/ValidateVirtualMethodAnnotationsStep.cs

```diff
@@ -2,13 +2,49 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics.CodeAnalysis;
+using Mono.Cecil;
+
 namespace Mono.Linker.Steps
 {
 	public class ValidateVirtualMethodAnnotationsStep : BaseStep
 	{
 		protected override void Process ()
 		{
-			Context.VirtualMethodAnnotationValidator.Validate ();
+			foreach (var method in Context.Annotations.VirtualMethodsWithAnnotationsToValidate) {
```

Comment:
You could store Context.Annotations in local variable to have fewer prefixes here